src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -692,4 +692,31 @@ export default class ReimbursementRequestService {
     if (!fileData) throw new NotFoundException('Image File', fileId);
     return fileData;
   }
+
+  /**
+   * Edits the vendor name
+   *
+   * @param name name to change of the vendor
+   * @param vendorId id of the vedor used to edit the name
+   * @param submitter the user who is downloading the receipt image
+   * @returns new vendor with a edited name
+   */
+  static async editVendors(name: string, vendorId: string, submitter: User) {
+    if (!isAdmin(submitter.role)) throw new AccessDeniedAdminOnlyException('only Admins can edit vendors');
+
+    const vendorExists = await prisma.vendor.findUnique({
+      where: { name }
+    });
+
+    console.log(vendorExists);
+
+    if (!!vendorExists) throw new HttpException(400, 'vendor name already exists');
+
+    const vendor = await prisma.vendor.update({
+      where: { vendorId },
+      data: { name }
+    });
+
+    return vendor;
```

Comment:
When we return the vendor object to the front end, we want to transform the type of it from `Prisma` to `Shared` type. To do this, we can wrap this vendor variable with `vendorTransformer`.